api/src/services/roles.ts

```diff
@@ -242,7 +247,7 @@ export class RolesService extends ItemsService {
 		for (const partialItem of data) {
 			this.assertValidIpAccess(partialItem);
 
-			if ('users' in partialItem) {
+			if ('users' in partialItem && needsUserLimitCheck) {
```

Comment:
I think it would be nicer if the "shouldCheckUserLimits" condition comes first in each of those `if`s, e.g.
```suggestion
			if (needsUserLimitCheck && 'users' in partialItem) {
```

Reply:
Sure thing 👍🏼